src/components/EditNoteForm/index.jsx

```diff
@@ -1,20 +1,27 @@
-import { useState, useRef } from 'react';
+import { useState, useRef, useEffect } from 'react';
 import { EditNoteFormPresentation } from './Presentation';
 import { useDispatch, useSelector } from 'react-redux';
 import { selectNotes, updateNote } from '../../app/slices/notesSlice';
 import { useParams } from 'react-router-dom';
+import { useGetNoteQuery } from '../../app/services/api';
 
 export const EditNoteForm = () => {
   const { noteId } = useParams();
-  const notes = useSelector(selectNotes);
-  const note = notes.find((note) => note.id === noteId);
+  const { data, isLoading } = useGetNoteQuery(noteId);
 
-  const [title, setTitle] = useState(note.title ?? '');
-  const [description, setDescription] = useState(note.description ?? '');
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
 
   const noteFormTitleRef = useRef(null);
   const noteFormDescriptionRef = useRef(null);
 
+  useEffect(() => {
+    if (!isLoading && data?.note) {
+      setTitle(data.note.title ?? '');
+      setDescription(data.note.description ?? '');
+    }
+  }, [data, isLoading]);
```

Comment:
these states are not needed, we can just use the data and isLoading value returned by the hook created by rtk query.